shell/platform/darwin/ios/framework/Source/FlutterEngineTest.mm

```diff
@@ -30,6 +31,24 @@ - (void)testCreate {
   XCTAssertNotNil(engine);
 }
 
+- (void)testInfoPlist {
+  // Check the embedded Flutter.framework Info.plist, not the linked dylib.
+  NSURL* flutterFrameworkURL =
+      [NSBundle.mainBundle.privateFrameworksURL URLByAppendingPathComponent:@"Flutter.framework"];
+  NSBundle* flutterBundle = [NSBundle bundleWithURL:flutterFrameworkURL];
+
+  NSDictionary<NSString*, id>* infoDictionary = flutterBundle.infoDictionary;
+  XCTAssertEqualObjects(infoDictionary[@"MinimumOSVersion"], @"9.0");
```

Comment:
`__ENVIRONMENT_IPHONE_OS_VERSION_MIN_REQUIRED__` maybe? Or, is the test harness not compiled with the same minimum OS version requirement? The next time we bump this up, we will need to update test too when all this test is doing is checking that what we specify in the GN rule ends up as verbatim in the Info.plist.

Reply:
>  Or, is the test harness not compiled with the same minimum OS version requirement?


It's not, the test lib is built with `ios_testing_deployment_target`
https://github.com/flutter/engine/blob/0292339eb1a970c5a70c38abb7ce836824c54019/shell/platform/darwin/ios/BUILD.gn#L228
And the test app is built with 12.2 
https://github.com/flutter/engine/blob/266435dafee3339283ecaed2b3bf1f34525c8748/testing/ios/IosUnitTests/IosUnitTests.xcodeproj/project.pbxproj#L374

> The next time we bump this up, we will need to update test too when all this test is doing is checking that what we specify in the GN rule ends up as verbatim in the Info.plist.

Well, it's also testing that the `copy_info_plist.py` script works.  I can just test that it's a well-formatted version string, if you prefer.  Otherwise, you're right, the buildroot SHA update that includes the config change will also need to update this test in the same commit.

Reply:
> I can just test that it's a well-formatted version string, if you prefer.

Don't have too strong of an opinion on this but I'd definitely prefer components and tests not making assumptions about the version floor. Just checking for a well formatted string should be fine.